langchain4j/src/main/java/dev/langchain4j/store/embedding/vespa/VespaEmbeddingStore.java

```diff
@@ -0,0 +1,180 @@
+    return (
+      "To use VespaEmbeddingStore, please add the following dependency to your project:\n\n" +
+      "Maven:\n" +
+      "<dependency>\n" +
+      "    <groupId>dev.langchain4j</groupId>\n" +
+      "    <artifactId>langchain4j-vespa</artifactId>\n" +
+      "    <version>0.21.0</version>\n" +
+      "</dependency>\n\n" +
+      "Gradle:\n" +
+      "implementation 'dev.langchain4j:langchain4j-vespa:0.21.0'\n"
+    );
+  }
+
+  private static EmbeddingStore<TextSegment> loadDynamically(
+    String implementationClassName,
+    String url,
+    String keyPath,
+    String certPath,
+    Duration timeout,
+    String namespace,
+    String documentType,
+    String rankProfile,
+    Integer targetHits,
+    Boolean avoidDups
+  )
+    throws ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
+    Class<?> implementationClass = Class.forName(implementationClassName);
+    Class<?>[] constructorParameterTypes = new Class<?>[] {
+      String.class,
+      String.class,
+      String.class,
+      Duration.class,
+      String.class,
+      String.class,
+      String.class,
+      Integer.class,
+      Boolean.class,
+    };
+    Constructor<?> constructor = implementationClass.getConstructor(constructorParameterTypes);
+    return (EmbeddingStore<TextSegment>) constructor.newInstance(
+      url,
+      keyPath,
+      certPath,
+      timeout,
+      namespace,
+      documentType,
+      rankProfile,
+      targetHits,
+      avoidDups
+    );
+  }
+
+  @Override
+  public String add(Embedding embedding) {
+    return implementation.add(embedding);
+  }
+
+  /**
+   * Adds a new embedding with provided ID to the store.
+   *
+   * @param id        "user-specified" part of document ID, find more details
+   *                  <a href="https://docs.vespa.ai/en/documents.html#namespace">here</a>
+   * @param embedding the embedding to add
+   */
+  @Override
+  public void add(String id, Embedding embedding) {
+    implementation.add(id, embedding);
+  }
+
+  @Override
+  public String add(Embedding embedding, TextSegment textSegment) {
+    return implementation.add(embedding, textSegment);
+  }
+
+  @Override
+  public List<String> addAll(List<Embedding> embeddings) {
+    return implementation.addAll(embeddings);
+  }
+
+  @Override
+  public List<String> addAll(List<Embedding> embeddings, List<TextSegment> textSegments) {
+    return implementation.addAll(embeddings, textSegments);
+  }
+
+  /**
+   * {@inheritDoc}
+   * The score inside {@link EmbeddingMatch} is Vespa relevance according to provided rank profile.
+   */
+  @Override
+  public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults) {
+    return implementation.findRelevant(referenceEmbedding, maxResults);
+  }
+
+  /**
+   * {@inheritDoc}
+   * The score inside {@link EmbeddingMatch} is Vespa relevance according to provided rank profile.
```

Comment:
I guess this doc will be outdated after adding convertations

Reply:
same here